File: ios/stream/stream/duration.swift
//
//  Duration.swift
//  stream
//
//  Created by xpwu on 2021/3/24.
//

import Foundation

public typealias Duration = UInt64

public extension Duration {
  static let Microsecond:UInt64 = 1
  static let Millisecond = 1000 * Microsecond
  static let Second = 1000 * Millisecond
  static let Minute = 60 * Second
  static let Hour = 60 * Minute
}

public extension Duration {
  func second() -> UInt64 {
    return self/Duration.Second
  }
  
  func milliSecond() -> UInt64 {
    return self/Duration.Millisecond
  }
  
  func minute() -> UInt64 {
    return self/Duration.Minute
  }
}

File: android/app/src/main/java/com/anywithyou/stream/duration.kt
@file:JvmName("Duration_Kt")

package com.anywithyou.stream

class DurationKt(internal val d: Long) {
	companion object {
		const val MicroSecond = 1L
		const val MilliSecond = 1000 * MicroSecond
		const val Second = 1000 * MilliSecond
		const val Minute = 60 * Second
		const val Hour = 60 * Minute
	}
}

val DurationKt.Second
	get() = d/DurationKt.Second

val DurationKt.MilliSecond
	get() = d/DurationKt.MilliSecond

val DurationKt.Minute
	get() = d/DurationKt.Minute
